src/Symfony/Component/Console/Application.php

```diff
@@ -167,8 +168,7 @@ public function doRun(InputInterface $input, OutputInterface $output)
         $name = $this->getCommandName($input);
         if (true === $input->hasParameterOption(array('--help', '-h'), true)) {
             if (!$name) {
-                $name = 'help';
-                $input = new ArrayInput(array('command' => 'help'));
+                $input = new ArrayInput(array('command_name' => $this->defaultCommand));
```

Comment:
I don't understand why you change this. the argument is named ``command``, not ``command_name`` in the Application